odoo/models.py

```diff
@@ -2349,6 +2349,12 @@ def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=Fals
                 # `False` instead of a (value, label) pair. In that case,
                 # leave the `False` value alone
                 if group.get(df):
+                    if self._fields[df.split(':')[0]].type == 'date':
+                        range_from, range_to = group[df][0].split('/')
+                        group["range"] = {
```

Comment:
FWIW this is a regular issue with groupby (it's really not well designed but there you go), this could be stored similar to the count / domain e.g. a `__range` key mapping to a sub-dict of `field: {from, to}`, something like that.